src/cache.ts

```diff
@@ -97,6 +98,7 @@ export class ListWatch<T extends KubernetesObject> implements ObjectCache<T>, In
             // do not auto-restart
             return;
         }
+        this.callbackCache[CONNECT].forEach((elt: ObjectCallback<T>) => elt({} as T));
```

Comment:
Is `{} as T` the right thing to do? I would expect no params passed to the callback here (not sure if TypeScript allows expressing that though...)

This also caught me off guard for the `ERROR` event - `error` is an instance of `Error`, not `T`.

Reply:
This was legacy left over from when the callbacks only dealt with objects not errors and connects. I refactored the callback type signature to make it better.